### ISSUE TITLE
Imported keys does not respect Signing/Encryption usage restriction -- Hanging on signing

### ISSUE THREAD

**Issue by bateast** [mailvelope/mailvelope#273]
I have two issue that might be related : 
1. Imported keys does not seem to respect Signing/Encryption usage restriction 
  on gnupg, my original keys are shown as
 ```
  gpg> list
  pub  rsa4096/DE838C7B
     created: 2015-01-12  expires: never       usage: SC  
  sub  rsa2048/0F72EFB8
     created: 2015-01-12  expires: 2015-07-11  usage: S   
  sub  rsa2048/641366EA
     created: 2015-01-12  expires: 2015-07-11  usage: E   
  ```
  and, after importing into mailvelope, both subkeys are shown as 
   ```RSA (Encrypt or Sign)```

2. With the previous imported key, though decryption works well, mailvelope hangs on key unlocking for signing after submiting either no, good or bad key password. 
No information is logged on console at all.

This behavior is not reproduced with a key generated by mailvelope.

(firefox 36.0, mailvelope 0.12.0b3)

**Comment by toberndo** [comment]
> and, after importing into mailvelope, both subkeys are shown as 
> RSA (Encrypt or Sign)

This value is taken from:
https://tools.ietf.org/html/rfc4880#section-9.1

Currently Mailvelope does not display the usage flag as in the gpg output.

> With the previous imported key, though decryption works well, mailvelope hangs on key unlocking 
> for signing after submiting either no, good or bad key password. No information is logged on 
> console at all.

Is this a recently generated key with GPG standard settings?

Thanks.

**Comment by bateast** [comment]
> Is this a recently generated key with GPG standard settings?

Well, I got the trick (and so reproduction steps)

Actually, I imported the private subkeys but not the private master key. It seems to be fine for decryption, but not enough for signing (actually, I don’t get why).

Steps to reproduces : 
1. generate a standard key pair with gpg
   ```$ gpg --gen-key```
   that would actually create a master key and a subkey. let default values (2048, no expire)
2. check it
   ```$ gpg -K```
    ```sec   rsa2048/0A7F3608 2015-02-04```
    ```uid       [ultimate] totototo```
    ```ssb   rsa2048/73FC54F9 2015-02-04```
3. export private **sub**keys
   ```$ gpg --armor --export-secret-subkey totototo > priv.asc```
4. import into mailvelope
5. try to sign a message with it
   
Thank you,
